### ISSUE TITLE
Default set inference_mode=True For saved checkpoint

### ISSUE THREAD

**Issue by dengchengxifrank** [huggingface/peft#2840]
### System Info

Package                  Version
------------------------ ------------------
accelerate               0.21.0
aiohappyeyeballs         2.4.0
aiohttp                  3.10.5
aiosignal                1.3.1
annotated-types          0.7.0
async-timeout            4.0.3
attrs                    24.2.0
audioread                3.0.1
bitsandbytes             0.42.0
certifi                  2024.8.30
cffi                     1.17.1
charset-normalizer       3.3.2
click                    8.1.7
cmake                    3.30.3
contourpy                1.3.0
cycler                   0.12.1
dataclasses              0.6
datasets                 2.21.0
decorator                5.1.1
dill                     0.3.6
einops                   0.8.1
evaluate                 0.4.2
filelock                 3.16.0
fonttools                4.54.1
frozenlist               1.4.1
fsspec                   2024.6.1
fvcore                   0.1.5.post20221221
hjson                    3.1.0
huggingface-hub          0.24.6
idna                     3.8
importlib_resources      6.4.5
iopath                   0.1.10
Jinja2                   3.1.4
jiwer                    3.0.4
joblib                   1.4.2
kaldiio                  2.18.0
kiwisolver               1.4.7
lazy_loader              0.4
librosa                  0.10.0
lit                      18.1.8
llvmlite                 0.43.0
MarkupSafe               2.1.5
matplotlib               3.9.2
mpmath                   1.3.0
msgpack                  1.1.0
multidict                6.1.0
multiprocess             0.70.14
networkx                 3.2.1
ninja                    1.11.1.4
numba                    0.60.0
numpy                    1.23.1
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-ml-py             12.575.51
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
packaging                24.1
pandas                   2.2.2
peft                     0.12.0
pillow                   10.4.0
pip                      24.2
platformdirs             4.3.2
pooch                    1.8.2
portalocker              3.2.0
protobuf                 5.28.0
psutil                   6.0.0
py-cpuinfo               9.0.0
pyarrow                  17.0.0
pycparser                2.22
pydantic                 2.11.7
pydantic_core            2.33.2
pyparsing                3.1.4
python-dateutil          2.9.0.post0
pytz                     2024.1
PyYAML                   6.0.2
rapidfuzz                3.9.7
regex                    2024.7.24
requests                 2.32.3
responses                0.18.0
safetensors              0.4.5
scikit-learn             1.5.1
scipy                    1.13.1
setuptools               72.1.0
six                      1.16.0
soundfile                0.12.1
soxr                     0.5.0.post1
sympy                    1.13.2
tabulate                 0.9.0
tensorboardX             2.6.2.2
termcolor                3.1.0
threadpoolctl            3.5.0
tokenizers               0.15.2
torch                    2.0.1
torchaudio               2.0.2
tqdm                     4.66.5
transformers             4.36.2
triton                   2.0.0
typing_extensions        4.12.2
typing-inspection        0.4.1
tzdata                   2024.1
urllib3                  2.2.2
wheel                    0.43.0
xxhash                   3.5.0
yacs                     0.1.8
yarl                     1.11.1
zhconv                   1.4.3
zipp                     3.20.2

### Who can help?

@BenjaminBossan @tmm1 

### Reproduction

I'm loading an already finetuned lora from the model_lora_path, and I want to continue training the lora parameters, so first I reload the checkpoint and I set inference_mode = False,init_lora_weights = False,is_trainable=True, after setting this and train the model I found that the new adapter_config.json file of the new checkpoint will always set inference_mode=True. 
I wonder two things:
1. Is my code correct? I mean based on my changes, can the lora first be load from model_lora_path and still trained for the new term?
2. I wonder why PEFT will re-write the inference_mode=True for checkpoint.

# The code is as follows:
```
peft_config = PeftConfig.from_pretrained(model_lora_path)
peft_config.modules_to_save = []
peft_config.inference_mode = False
peft_config.init_lora_weights = False

model=PeftModel.from_pretrained(model,model_lora_path,config=peft_config,is_trainable=True,ignore_mismatched_sizes=True).cuda()

for adapter_name in model.peft_config.keys():
    pdb.set_trace()
    model.peft_config[adapter_name].inference_mode = False
    print(f"Adapter '{adapter_name}' inference_mode: {model.peft_config[adapter_name].inference_mode}")
```

### Expected behavior

Maybe set peft_config.inference_mode as how user set for training is better？

**Comment by BenjaminBossan** [comment]
The way you load the model by first loading the config and then the model is highly unusual, do you have a specific reason to do that?

Normally, when I want to load a checkpoint and continue training, I would just do:

```python
base_model = ...
peft_model = PeftModel.from_pretrained(base_model, model_lora_path, is_trainable=True)
```

You don't really have to bother with what is set in the config, if you pass `is_trainable=True`, it should work.

**Comment by dengchengxifrank** [comment]
I get it. Thanks!
